examples/tti_operators.py

```diff
@@ -0,0 +1,219 @@
+from devito.operators import *
+from sympy import Eq
+from devito.interfaces import TimeData, DenseData
+from sympy import Function, symbols, as_finite_diff, Wild
+from sympy.abc import x, y, t, z
+from sympy import solve
+from sympy import *
+from sympy.abc import *
+from fwi_operators import SourceLike
+
+
+class SourceLikeTTI(SourceLike):
+    def read(self, u, v):
+        eqs = []
+        for i in range(self.npoints):
+            eqs.append(Eq(self[t, i], (self.grid2point(u, self.orig_data[i, :]) + self.grid2point(v, self.orig_data[i, :]))/2))
+        return eqs
+
+
+class TTIOperator(Operator):
+    def _init_taylor(self, dim=2, time_order=2, space_order=2):
+        # Only dim=2 and dim=3 are supported
+        # The acoustic wave equation for the square slowness m and a source q
+        # is given in 3D by :
+        #
+        # \begin{cases} &m \frac{d^2 u(x,t)}{dt^2} - \nabla^2 u(x,t) =q  \\
+        # &u(.,0) = 0 \\ &\frac{d u(x,t)}{dt}|_{t=0} = 0 \end{cases}
+        #
+        # with the zero initial conditons to guarantee unicity of the solution
+        # Choose dimension (2 or 3)
+        # half width for indexes, goes from -half to half
+        p = Function('p')
+        r = Function('r')
+
+        s, h, x, y, z = symbols('s h x y z')
+        m = M(x, y, z)
+        # q = Q(x, y, z, t)
+        # d = D(x, y, z, t)
+        e = E(x, y, z)
+
+        A = epsilon(x, y, z)  # (1 + 2epsilon) but make the symbolic simpler
+        B = delta(x, y, z)  # sqrt(1 + 2epsilon) but make the symbolic simpler
+        Th = theta(x, y, z)
+        Ph = phi(x, y, z)
+        # Weights to sum the two fields
+        # w1 = .5
+        # w2 = .5
+        dttp = as_finite_diff(p(x, y, z, t).diff(t, t), [t-s, t, t+s])
+        dttr = as_finite_diff(r(x, y, z, t).diff(t, t), [t-s, t, t+s])
+        # dtp = as_finite_diff(p(x, y, z, t).diff(t), [t-s, t])
+        # dtr = as_finite_diff(r(x, y, z, t).diff(t), [t-s, t])
```

Comment:
These comments look like debug/dev leftovers. Are they necessary? Same goes for the symbol and parameter definitions above.